django/db/models/fields/__init__.py

```diff
@@ -1963,6 +1963,8 @@ def get_internal_type(self):
         return "GenericIPAddressField"
 
     def to_python(self, value):
+        if value and not isinstance(value, six.string_types):
+            value = smart_text(value)
```

Comment:
I'd think we may need to use `force_text()` -- won't the next line fail if a lazy instance isn't resolved?